Fix the problem in mips and mipsel about printing the value of
'complex(NaN,0)'.  This was causing the package to FTBFS when
running the test of function log2 in data.cc (fixes Bug#532656).

 -- Rafael Laboissiere <rafael@debian.org>  Mon, 15 Jun 2009 12:56:53 +0200

--- a/src/pr-output.cc
+++ b/src/pr-output.cc
@@ -853,10 +853,12 @@
   double i_abs = ip < 0.0 ? -ip : ip;
 
   int r_x = r_abs == 0.0
-    ? 0 : static_cast<int> (floor (log10 (r_abs) + 1.0));
+    ? 0 : ((xisinf (rp) || xisnan (rp))
+	   ? INT_MIN : static_cast<int> (floor (log10 (r_abs) + 1.0)));
 
   int i_x = i_abs == 0.0
-    ? 0 : static_cast<int> (floor (log10 (i_abs) + 1.0));
+    ? 0 : ((xisinf (ip) || xisnan (ip))
+	   ? INT_MIN : static_cast<int> (floor (log10 (i_abs) + 1.0)));
 
   int x_max, x_min;
 
